collector_test.go

```diff
@@ -0,0 +1,321 @@
+package redigosrv
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/gomodule/redigo/redis"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/prometheus/client_golang/prometheus"
+	dto "github.com/prometheus/client_model/go"
+)
+
+type poolStatsFake struct{}
+
+func (psf *poolStatsFake) Stats() redis.PoolStats {
+	return redis.PoolStats{
+		ActiveCount: 10,
+		IdleCount:   7,
+	}
+}
+
+var _ = Describe("RedigoCollector", func() {
+
+	var service RedigoService
+	var metric dto.Metric
+
+	BeforeEach(func() {
+		Expect(service.ApplyConfiguration(Configuration{
+			Address: "localhost:6379",
+		})).To(BeNil())
+		Expect(service.Start()).To(BeNil())
+	})
+
+	It("should count total of data send using method Publish", func(done Done) {
+
+		defer service.Stop()
```

Comment:
You can stop service in an `AfterEach` function (as the same way that you did starting service)

Reply:
Solved, Mr. :grin: